src/restic/node.go

```diff
@@ -331,11 +321,26 @@ func (node *Node) UnmarshalJSON(data []byte) error {
 	type nodeJSON Node
 	nj := (*nodeJSON)(node)
 
-	err := json.Unmarshal(data, nj)
-	if err != nil {
-		return errors.Wrap(err, "Unmarshal")
+	if err := json.Unmarshal(data, nj); err != nil {
+		// Retry conversion with list of ExtendedAttributes
```

Comment:
Hm, now we have a slightly nicer structure and the comparison code is shorter, but `Unmarshal` is way more complex. I must admit that I do not really like it.